### ISSUE TITLE
[HOLD for payment 2023-03-06] [$1000] IOU details : Last transaction appears at the bottom in the list until logout and login back

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#14684]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open the app and login with user A (Any device)
2. On second device, keep user B logged in (web chrome)
3. Open report of user B on user A logged in device
4. Request money from user B
5. Check the latest transaction position on user B iou details page

## Expected Result:
New transactions should be displayed on the top

## Actual Result:
Transactions are displayed in the bottom and sequence is only fixed when we log out and again log in

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** v1.2.62-1
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:**  

https://user-images.githubusercontent.com/43996225/215642457-079cd40f-f3a5-44d9-98e1-8731ad744244.mp4

https://user-images.githubusercontent.com/43996225/215642469-77e9708c-57ee-4716-b16f-e83328fcd816.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @dhanashree-sawant
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1674836629400269

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01a8808fbd0d76ee31</li>
        <li>Upwork Job ID: 1623391957221736448</li>
        <li>Last Price Increase: 2023-03-07</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
@tjferriss Still overdue 6 days?! Let's take care of this!

**Comment by tjferriss** [comment]
This one got lost in the shuffle. I was not able to reproduce it. I've asked in [Slack](https://expensify.slack.com/archives/C049HHMV9SM/p1675837867037309?thread_ts=1674836629.400269&cid=C049HHMV9SM) whether this is still occurring.

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01a8808fbd0d76ee31

**Comment by MelvinBot** [comment]
Current assignee @tjferriss is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @mollfpr (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @tgolen (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by tienifr** [comment]
# Proposal
## Problem
In the IOUTransaction component, we **don't sort** and we also don't filter out invalid report actions (that have sequenceNumber field) to prevent bugs during the migration from sequenceNumber -> reportActionID like the `ReportActionView` Component
## Solution
In /Expensify/App/src/pages/iou/IOUTransactions.js
We should add a function to sort and filter out report actions like [ReportActionView Component](https://github.com/Expensify/App/blob/8f754df0ddea4249eb5d1f94ea6642a8a797d08c/src/pages/home/report/ReportActionsView.js#L258)



```diff --git a/src/libs/ReportActionsUtils.js b/src/libs/ReportActionsUtils.js
index bb67b2fb39..2b1ca41b6e 100644
--- a/src/libs/ReportActionsUtils.js
+++ b/src/libs/ReportActionsUtils.js
@@ -98,6 +98,34 @@ function filterReportActionsForDisplay(reportActions) {
     });
 }

+/**
+ * Filter out any reportActions which should not be displayed
+ * Then, sort by their created timestamp first, and reportActionID second
+ * This is a combination of getSortedReportActions() and filterReportActionsForDisplay()
+ *
+ * @param {Array} reportActions
+ * @returns {Array}
+ */
+function getSortedReportActionsForDisplay(reportActions) {
+    // HACK ALERT: We're temporarily filtering out any reportActions keyed by sequenceNumber
+    // to prevent bugs during the migration from sequenceNumber -> reportActionID
+    const filteredReportActions = _.filter(reportActions, (reportAction, key) => {
+        if (!reportAction) {
+            return false;
+        }
+
+        if (String(reportAction.sequenceNumber) === key) {
+            return false;
+        }
+
+        return true;
+    });
+
+    let sortedReportActions = getSortedReportActions(filteredReportActions, true);
+    sortedReportActions = filterReportActionsForDisplay(sortedReportActions);
+    return sortedReportActions;
+}
+
 /**
  * Finds most recent IOU report action number.
  *
@@ -176,6 +204,7 @@ function getLastVisibleMessageText(reportID, actionsToMerge = {}) {
 export {
     getSortedReportActions,
     filterReportActionsForDisplay,
+    getSortedReportActionsForDisplay,
     getLastVisibleAction,
     getLastVisibleMessageText,
     getMostRecentIOUReportActionID,
diff --git a/src/pages/home/report/ReportActionsView.js b/src/pages/home/report/ReportActionsView.js
index 4bbeaa997..878287099 100755
--- a/src/pages/home/report/ReportActionsView.js
+++ b/src/pages/home/report/ReportActionsView.js
@@ -63,7 +63,7 @@ class ReportActionsView extends React.Component {
         this.hasCachedActions = _.size(props.reportActions) > 0;
 
         // We need this.sortedAndFilteredReportActions to be set before this.state is initialized because the function to calculate the newMarkerReportActionID uses the sorted report actions
-        this.sortedAndFilteredReportActions = this.getSortedReportActionsForDisplay(props.reportActions);
+        this.sortedAndFilteredReportActions = ReportActionsUtils.getSortedReportActionsForDisplay(props.reportActions);
 
         this.state = {
             isFloatingMessageCounterVisible: false,
@@ -128,7 +128,7 @@ class ReportActionsView extends React.Component {
 
     shouldComponentUpdate(nextProps, nextState) {
         if (!_.isEqual(nextProps.reportActions, this.props.reportActions)) {
-            this.sortedAndFilteredReportActions = this.getSortedReportActionsForDisplay(nextProps.reportActions);
+            this.sortedAndFilteredReportActions = ReportActionsUtils.getSortedReportActionsForDisplay(nextProps.reportActions);
             this.mostRecentIOUReportActionID = ReportActionsUtils.getMostRecentIOUReportActionID(nextProps.reportActions);
             return true;
         }
@@ -251,29 +251,6 @@ class ReportActionsView extends React.Component {
         Report.unsubscribeFromReportChannel(this.props.report.reportID);
     }
 
-    /**
-     * @param {Object} reportActions
-     * @returns {Array}
-     */
-    getSortedReportActionsForDisplay(reportActions) {
-        // HACK ALERT: We're temporarily filtering out any reportActions keyed by sequenceNumber
-        // to prevent bugs during the migration from sequenceNumber -> reportActionID
-        const filteredReportActions = _.filter(reportActions, (reportAction, key) => {
-            if (!reportAction) {
-                return false;
-            }
-
-            if (String(reportAction.sequenceNumber) === key) {
-                return false;
-            }
-
-            return true;
-        });
-
-        const sortedReportActions = ReportActionsUtils.getSortedReportActions(filteredReportActions, true);
-        return ReportActionsUtils.filterReportActionsForDisplay(sortedReportActions);
-    }
-
     /**
diff --git a/src/pages/iou/IOUTransactions.js b/src/pages/iou/IOUTransactions.js
index 7b540336c3..836a8e38ac 100644
--- a/src/pages/iou/IOUTransactions.js
+++ b/src/pages/iou/IOUTransactions.js
@@ -8,6 +8,7 @@ import styles from '../../styles/styles';
 import ONYXKEYS from '../../ONYXKEYS';
 import reportActionPropTypes from '../home/report/reportActionPropTypes';
 import ReportTransaction from '../../components/ReportTransaction';
+import * as ReportActionsUtils from '../../libs/ReportActionsUtils';
 import CONST from '../../CONST';

 const propTypes = {
@@ -70,9 +71,10 @@ class IOUTransactions extends Component {
     }

     render() {
+        const sortedReportActions = ReportActionsUtils.getSortedReportActionsForDisplay(this.props.reportActions)
         return (
             <View style={[styles.mt3]}>
-                {_.map(this.props.reportActions, (reportAction) => {
+                {_.map(sortedReportActions, (reportAction) => {
                     // iouReportIDs should be strings, but we still have places that send them as ints so we convert them both to Numbers for comparison
                     if (!reportAction.originalMessage || Number(reportAction.originalMessage.IOUReportID) !== Number(this.props.iouReportID)) {
                         return;

```



## Result 

https://user-images.githubusercontent.com/113963320/216879512-8eedb9ba-1158-4193-8338-ef7bce50333b.mp4

**Comment by tienifr** [comment]
I update [the earlier proposal](https://github.com/Expensify/App/issues/14842#issuecomment-1423291114) to follow the recommended template

# Updated Proposal
## Please re-state the problem that we are trying to solve in this issue.
The new transaction always display at the bottom of the transaction list until logout and login back 
## What is the root cause of that problem?
In the IOUTransaction component, we **don't sort** and we also don't filter out invalid report actions (that have sequenceNumber field) to prevent bugs during the migration from sequenceNumber -> reportActionID like the `ReportActionView` Component

## What changes do you think we should make in order to solve the problem?
In /Expensify/App/src/pages/iou/IOUTransactions.js
We should add a function to sort and filter out report actions like [ReportActionView Component](https://github.com/Expensify/App/blob/8f754df0ddea4249eb5d1f94ea6642a8a797d08c/src/pages/home/report/ReportActionsView.js#L258)



```diff 
--git a/src/libs/ReportActionsUtils.js b/src/libs/ReportActionsUtils.js
index bb67b2fb39..2b1ca41b6e 100644
--- a/src/libs/ReportActionsUtils.js
+++ b/src/libs/ReportActionsUtils.js
@@ -98,6 +98,34 @@ function filterReportActionsForDisplay(reportActions) {
     });
 }

+/**
+ * Filter out any reportActions which should not be displayed
+ * Then, sort by their created timestamp first, and reportActionID second
+ * This is a combination of getSortedReportActions() and filterReportActionsForDisplay()
+ *
+ * @param {Array} reportActions
+ * @returns {Array}
+ */
+function getSortedReportActionsForDisplay(reportActions) {
+    // HACK ALERT: We're temporarily filtering out any reportActions keyed by sequenceNumber
+    // to prevent bugs during the migration from sequenceNumber -> reportActionID
+    const filteredReportActions = _.filter(reportActions, (reportAction, key) => {
+        if (!reportAction) {
+            return false;
+        }
+
+        if (String(reportAction.sequenceNumber) === key) {
+            return false;
+        }
+
+        return true;
+    });
+
+    let sortedReportActions = getSortedReportActions(filteredReportActions, true);
+    sortedReportActions = filterReportActionsForDisplay(sortedReportActions);
+    return sortedReportActions;
+}
+
 /**
  * Finds most recent IOU report action number.
  *
@@ -176,6 +204,7 @@ function getLastVisibleMessageText(reportID, actionsToMerge = {}) {
 export {
     getSortedReportActions,
     filterReportActionsForDisplay,
+    getSortedReportActionsForDisplay,
     getLastVisibleAction,
     getLastVisibleMessageText,
     getMostRecentIOUReportActionID,
diff --git a/src/pages/home/report/ReportActionsView.js b/src/pages/home/report/ReportActionsView.js
index 4bbeaa997..878287099 100755
--- a/src/pages/home/report/ReportActionsView.js
+++ b/src/pages/home/report/ReportActionsView.js
@@ -63,7 +63,7 @@ class ReportActionsView extends React.Component {
         this.hasCachedActions = _.size(props.reportActions) > 0;
 
         // We need this.sortedAndFilteredReportActions to be set before this.state is initialized because the function to calculate the newMarkerReportActionID uses the sorted report actions
-        this.sortedAndFilteredReportActions = this.getSortedReportActionsForDisplay(props.reportActions);
+        this.sortedAndFilteredReportActions = ReportActionsUtils.getSortedReportActionsForDisplay(props.reportActions);
 
         this.state = {
             isFloatingMessageCounterVisible: false,
@@ -128,7 +128,7 @@ class ReportActionsView extends React.Component {
 
     shouldComponentUpdate(nextProps, nextState) {
         if (!_.isEqual(nextProps.reportActions, this.props.reportActions)) {
-            this.sortedAndFilteredReportActions = this.getSortedReportActionsForDisplay(nextProps.reportActions);
+            this.sortedAndFilteredReportActions = ReportActionsUtils.getSortedReportActionsForDisplay(nextProps.reportActions);
             this.mostRecentIOUReportActionID = ReportActionsUtils.getMostRecentIOUReportActionID(nextProps.reportActions);
             return true;
         }
@@ -251,29 +251,6 @@ class ReportActionsView extends React.Component {
         Report.unsubscribeFromReportChannel(this.props.report.reportID);
     }
 
-    /**
-     * @param {Object} reportActions
-     * @returns {Array}
-     */
-    getSortedReportActionsForDisplay(reportActions) {
-        // HACK ALERT: We're temporarily filtering out any reportActions keyed by sequenceNumber
-        // to prevent bugs during the migration from sequenceNumber -> reportActionID
-        const filteredReportActions = _.filter(reportActions, (reportAction, key) => {
-            if (!reportAction) {
-                return false;
-            }
-
-            if (String(reportAction.sequenceNumber) === key) {
-                return false;
-            }
-
-            return true;
-        });
-
-        const sortedReportActions = ReportActionsUtils.getSortedReportActions(filteredReportActions, true);
-        return ReportActionsUtils.filterReportActionsForDisplay(sortedReportActions);
-    }
-
     /**
diff --git a/src/pages/iou/IOUTransactions.js b/src/pages/iou/IOUTransactions.js
index 7b540336c3..836a8e38ac 100644
--- a/src/pages/iou/IOUTransactions.js
+++ b/src/pages/iou/IOUTransactions.js
@@ -8,6 +8,7 @@ import styles from '../../styles/styles';
 import ONYXKEYS from '../../ONYXKEYS';
 import reportActionPropTypes from '../home/report/reportActionPropTypes';
 import ReportTransaction from '../../components/ReportTransaction';
+import * as ReportActionsUtils from '../../libs/ReportActionsUtils';
 import CONST from '../../CONST';

 const propTypes = {
@@ -70,9 +71,10 @@ class IOUTransactions extends Component {
     }

     render() {
+        const sortedReportActions = ReportActionsUtils.getSortedReportActionsForDisplay(this.props.reportActions)
         return (
             <View style={[styles.mt3]}>
-                {_.map(this.props.reportActions, (reportAction) => {
+                {_.map(sortedReportActions, (reportAction) => {
                     // iouReportIDs should be strings, but we still have places that send them as ints so we convert them both to Numbers for comparison
                     if (!reportAction.originalMessage || Number(reportAction.originalMessage.IOUReportID) !== Number(this.props.iouReportID)) {
                         return;

```



## Result



https://user-images.githubusercontent.com/113963320/216879512-8eedb9ba-1158-4193-8338-ef7bce50333b.mp4

**Comment by tgolen** [comment]
@tienifr Can you please rewrite your proposal to follow this instructions in the proposal template? https://github.com/Expensify/App/blob/main/contributingGuides/PROPOSAL_TEMPLATE.md 

Primarily, this part:
```
<!-- DO NOT POST CODE DIFFS -->
```

**Comment by tienifr** [comment]
# Proposal
## Please re-state the problem that we are trying to solve in this issue.
The new transaction always display at the bottom of the transaction list until logout and login back 
## What is the root cause of that problem?
In the IOUTransaction component, we **don't sort** by timestamp (the `created` field in reportActions) and we also don't filter out invalid report actions (that have sequenceNumber field) to prevent bugs during the migration from sequenceNumber -> reportActionID like the `ReportActionView` Component

## What changes do you think we should make in order to solve the problem?
First, we need to create a util function to sort and filter reportActions. The function is already there in [ReportActionsView Component](https://github.com/Expensify/App/blob/d9a126dba03ea2956176e33ff1c6d869a2653d03/src/pages/home/report/ReportActionsView.js#L258-L275), we should move it to ReportActionsUtils for reuse. Then we can use that function inside ReportActionsView Component and IOUTransactions Component. In /App/src/pages/iou/IOUTransactions.js, we use that function in render function like below
```
render() {
      const sortedReportActions = ReportActionsUtils.getSortedReportActionsForDisplay(this.props.reportActions)
...
```
and then using sortedReportActions instead of this.props.reportActions

## Result

https://user-images.githubusercontent.com/113963320/216879512-8eedb9ba-1158-4193-8338-ef7bce50333b.mp4

**Comment by mollfpr** [comment]
> we also don't filter out invalid report actions (that have sequenceNumber field) to prevent bugs during the migration from sequenceNumber -> reportActionID like the ReportActionView Component

@tienifr Could you explain more about the bug you mention?

@tgolen The solution seems logically correct to me. It makes sense to have the sort of according to the _created_ timestamp of the report action.

**Comment by tienifr** [comment]
@mollfpr The bug occured because in the IOUTransaction component, the `reportActions` variable is not sorted yet. That's the primary issue.

What I meant is by reusing the sort function in  https://github.com/Expensify/App/blob/d9a126dba03ea2956176e33ff1c6d869a2653d03/src/pages/home/report/ReportActionsView.js#L258-L275
 for the above `reportActions` variable, it will miltigate the issue.

**Comment by tienifr** [comment]
@mollfpr  
> to prevent bugs during the migration from sequenceNumber -> reportActionID 

before, reportActions are keyed by sequenceNumber and we are migrating to reportActionId. I believe that we need to filter out invalid report actions (that have sequenceNumber field) because in some cases when there is a new report action, 2 report actions are created (one with a key of sequenceNumber one with a key of reportActionId) and the hack was added in [this PR](https://github.com/Expensify/App/pull/13118). And [this commit](https://github.com/Expensify/App/commit/d8dd9ce4d1e18e259ba020d879a654bba84996af) noted the comment in code
`
// HACK ALERT: We're temporarily filtering out any reportActions keyed by sequenceNumber 
`
`
// to prevent bugs during the migration from sequenceNumber -> reportActionID 
`
cc @roryabraham to confirm

**Comment by tgolen** [comment]
> @tgolen The solution seems logically correct to me. It makes sense to have the sort of according to the created timestamp of the report action.

I agree. I think this proposal makes sense and to abstract a sorting function into a shared utility method. I think that hack also needs to stay for now and we should have an issue at some point to come back and clean that up.

**Comment by roryabraham** [comment]
There is an issue to remove that hack [here](https://github.com/Expensify/App/issues/14574) (just trying to be cautious before removing it)

**Comment by tgolen** [comment]
Sorry, I'm not exactly sure what you are implying. It seems to me, that if there are two separate components, that have the exact same sort function, then we should DRY that code up into a utility file so the same sort method can be used for both components.

**Comment by mollfpr** [comment]
I'm sorry, I don't understand the intention of the solution. I thought that we will remove the hack in this issue, but I think that we still need the hack. Do you think we should continue with the assignment?

**Comment by tgolen** [comment]
OK, I think "the hack" is not touched in this solution at all and it needs to remain in the code for now. I don't think anyone is suggesting that "the hack" gets removed in this solution. Sorry if you got that impression from something I said. So, I think you and I are both on the same page. 

There will be another PR that removes the hack which is what [this issue](https://github.com/Expensify/App/issues/14574) is for.

**Comment by mollfpr** [comment]
Sounds good to me!

**Comment by MelvinBot** [comment]
@tgolen @tjferriss @mollfpr this issue was created 2 weeks ago. Are we close to approving a proposal? If not, what's blocking us from getting this issue assigned? Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by tgolen** [comment]
@MelvinBot it sounds like we are agreed on the proposal and @tienifr has the 🟢  to proceed.

**Comment by tienifr** [comment]
@tgolen can you assign me to the issue? Similar to [this](https://github.com/Expensify/App/issues/14842#issuecomment-1426040431), so I can start preparing the PR.

Thanks!

**Comment by tienifr** [comment]
Hi @tgolen @mollfpr This PR](https://github.com/Expensify/App/pull/15160) is ready for review

**Comment by tienifr** [comment]
Hi @mollfpr Could you take a look at this PR, Thanks.

**Comment by tienifr** [comment]
Hi @tgolen [the PR](https://github.com/Expensify/App/pull/15160) is ready for review and got @mollfpr's approval. Please help me check it. Thank you!

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.2.76-7 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/15160


If no regressions arise, payment will be issued on 2023-03-06. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@mollfpr / @tgolen] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@mollfpr / @tgolen] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@mollfpr / @tgolen] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@tienifr] Propose [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@tjferriss] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by tienifr** [comment]
# Regression Test Proposal

**Bug**: IOU details: The last transaction appears at the bottom in the list until logout and login back

**Proposed Test Steps**:

1. Open the app and log in with user A
2. Open the app and log in with user B on another device
3. On user A request money from user B two times any amount
4. On user B check the IOU action box and make sure that the latest amount request is showing
5. Still on user B, open the IOU detail by pressing the IOU action box
6. Verify the last transaction amount requested from user A is on the top of the list of transaction


Do we :+1: or :-1:

**Comment by mollfpr** [comment]
@tgolen @tjferriss I couldn't find the regression source except that this happened due to Onyx migration data, and we are not using `sequenceNumbers` number anymore for the keys based on this PR https://github.com/Expensify/App/pull/13118. Do we agree?

> [@mollfpr / @tgolen] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

I think there's no need to update the checklist.

**Comment by mollfpr** [comment]
@tjferriss Applied!

**Comment by tienifr** [comment]
@tjferriss Thanks, I've applied.

**Comment by dhanashree-sawant** [comment]
@tjferriss Thanks, I have also applied.

**Comment by MelvinBot** [comment]
Upwork job price has been updated to $1000

**Comment by tgolen** [comment]
@tjferriss Looks like you've got the next steps on this one?

**Comment by tjferriss** [comment]
Everyone has been paid in Upworks.

**Comment by tjferriss** [comment]
@tgolen could you help me out with @mollfpr comment [here](https://github.com/Expensify/App/issues/14684#issuecomment-1456365704). 

Based on this feedback, do you still think we need to setup a regression test? I'm a bit lost in the technicals.

**Comment by tgolen** [comment]
@tjferriss Is this good? Do you have everything you need now?

**Comment by tjferriss** [comment]
I've created the issue for the regression test. I believe we're all set now.
